crates/hstr/src/tagged_value.rs

```diff
@@ -73,7 +73,7 @@ impl TaggedValue {
     pub fn new_tag(value: NonZeroU8) -> Self {
         let value = value.get() as RawTaggedValue;
         Self {
-            value: unsafe { std::mem::transmute(value) },
+            value: unsafe { std::mem::transmute::<usize, std::ptr::NonNull<()>>(value) },
```

Comment:
I'm not familiar with rust, but i get the following error when building wasm:

```bash
error[E0308]: mismatched types
    --> /Users/username/.cargo/git/checkouts/ddbase-50741f53ff8906a6/ff40b64/crates/hstr/src/tagged_value.rs:76:81
     |
76   |             value: unsafe { std::mem::transmute::<usize, std::ptr::NonNull<()>>(value) },
     |                             --------------------------------------------------- ^^^^^ expected `usize`, found `u64`
     |                             |
     |                             arguments to this function are incorrect
     |
note: function defined here
    --> /Users/username/.rustup/toolchains/nightly-2024-08-01-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/intrinsics.rs:1389:12
     |
1389 |     pub fn transmute<Src, Dst>(src: Src) -> Dst;
     |            ^^^^^^^^^
help: you can convert a `u64` to a `usize` and panic if the converted value doesn't fit
     |
76   |             value: unsafe { std::mem::transmute::<usize, std::ptr::NonNull<()>>(value.try_into().unwrap()) },
     |                                                                                      ++++++++++++++++++++

error[E0308]: mismatched types
  --> /Users/username/.cargo/git/checkouts/ddbase-50741f53ff8906a6/ff40b64/crates/hstr/src/tagged_value.rs:76:29
   |
76 |             value: unsafe { std::mem::transmute::<usize, std::ptr::NonNull<()>>(value) },
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `NonZero<u64>`, found `NonNull<()>`
   |
   = note: expected struct `NonZero<u64>`
              found struct `NonNull<()>`

For more information about this error, try `rustc --explain E0308`.
```

Reply:
I try switching to version `0.2.10`, it worked

Reply:
Fixed it. Thank you for reporting!